ios/Handlers/ChangePinHandler.swift

```diff
@@ -0,0 +1,45 @@
+protocol PinConnectorToPinHandler: AnyObject {
+    func onChangePinCalled(completion: @escaping (Bool, NSError?) -> Void)
+}
+
+enum PINEntryMode {
+    case login
+    case registration
+    
+}
+
+class ChangePinHandler: NSObject {
+    var flow: PinFlow?
+    var changePinCompletion: ((Bool, NSError?) -> Void)?
+    let pinAuthenticationEventEmitter = PinAuthenticationEventEmitter()
+    let createPinEventEmitter = CreatePinEventEmitter()
+}
+
+extension ChangePinHandler : PinConnectorToPinHandler {
+    func onChangePinCalled(completion: @escaping (Bool, NSError?) -> Void) {
+        ONGUserClient.sharedInstance().changePin(self)
+        changePinCompletion = completion
+    }
+ }
+
+extension ChangePinHandler: ONGChangePinDelegate {
+    func userClient(_ userClient: ONGUserClient, didReceive challenge: ONGPinChallenge) {
+        BridgeConnector.shared?.toLoginHandler.handleDidReceiveChallenge(challenge)
```

Comment:
create a computed property for `BridgeConnector.shared?` and use it everywhere here